client/src/main/java/com/streamr/client/rest/Json.java

```diff
@@ -16,4 +20,12 @@ private Json() {}
         .add(Address.class, new AddressJsonAdapter().nullSafe())
         .add(new InstantJsonAdapter());
   }
+
```

Comment:
One of the issues I encountered while reading Java client code was that all JSON operations used same JSON class even though they didn't need all adapters. That made understanding the code harder than necessary.

Instantiating new Moshi adapters is practically free. So it makes sense to instantiate one and forget about it. Also note that Adapters are not Thread safe so storing an adapter in a `static` variable is out of the question.

I'll ned to add a comment to method (above) `newMoshiBuilder()` that explains this or inline all calls and remove it.

Reply:
Hmm didn't quite understand. Is there something that needs to be changed in this PR?